### ISSUE TITLE
Update reader read methods to include pre/post read checks

### ISSUE THREAD

**Issue by drewejohnson** [CORE-GATECH-GROUP/serpent-tools#86]
Implement reader-specific methods for inspections before and after reading. Pre-checks could include ensuring the file exists, simple validation that the file is of the correct type (e.g. not reading a detector file with a depletion reader), or counting the number of time steps present in the results file #7. Post-read checks could include verifying that data was indeed stored. If nothing is stored, Exceptions should be raised, perhaps a new one like `EmptyReader` that inherits from `SerpentToolsException`

In order to accomplish this, I propose refactoring the public read methods for all readers to private `_read` methods. The base reader will contain the public read method, and seek for precheck and postcheck methods specific to each file type. These do not have to be overwritten for each reader. 

## Implementation Suggestion
```
class BaseReader(object):

    def read(self):
        self._precheck()
        # maybe some debug/info statements
        self._read()
        self._postcheck()
    def _precheck():
         pass
    def _postcheck():
        pass
    def _read():
        raise NotImplementedError

class SubClass(BaseReader):

    def _read():
        # do the things
    def _postcheck():
        assert thingsWereDone
````
Pre/post check functions should [log errors/warnings](https://github.com/CORE-GATECH-GROUP/serpent-tools/blob/master/serpentTools/messages.py) accordingly, such as an error if the word `DET` does not appear in the first fuel lines of a supposed detector file. When applicable, Exceptions should be raised if things go really poorly.

**Comment by gridley** [comment]
I tried parsing a COE file the other day which only finished partially. Seems that this doesn’t get caught as-is.

**Comment by drewejohnson** [comment]
Agreed. I read a handful of depletion files as detectors and nothing was mentioned/raised 🙁 This will definitely help

**Comment by gridley** [comment]
I would absolutely be willing to take this issue, if you want.

**Comment by drewejohnson** [comment]
Sounds great! All yours

**Comment by drewejohnson** [comment]
Closed with #88
